src/model/error/UnexpectedNumberOfMessagesError.ts

```diff
@@ -0,0 +1,12 @@
+export class UnexpectedNumberOfMessagesError extends Error {
+    constructor(
+        numberOfReceivedMessages?: number,
+        expectedNumberOfMessages?: number,
+    ) {
+        super(
+            `Received an unexpected number of messages: ${numberOfReceivedMessages} (expected: ${expectedNumberOfMessages})`,
+        );
+        Object.setPrototypeOf(this, new.target.prototype);
```

Comment:
Maybe add a comment why we are doing this here.
Either a comment that the prototype chain is broken & it has to be restored manually ( as mentioned here https://stackoverflow.com/a/48342359 ) 

or a link to the the recommendation here (preferred)
https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work